scheduler/config/plugin.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"golang.org/x/xerrors"
+	"k8s.io/kube-scheduler/config/v1beta2"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins"
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+)
+
+// RegisteredFilterPlugins returns all registered plugins.
+// in-tree plugins and your original plugins listed in AdditionalFilterPlugins.
+func RegisteredFilterPlugins() ([]v1beta2.Plugin, error) {
+	def, err := DefaultFilterPluginSet()
+	if err != nil {
+		return nil, xerrors.Errorf("get default filter plugins: %w", err)
+	}
+
+	return append(def.Enabled, AdditionalFilterPlugins()...), nil
+}
+
+func DefaultFilterPluginSet() (v1beta2.PluginSet, error) {
+	defaultConfig, err := DefaultSchedulerConfig()
+	if err != nil || len(defaultConfig.Profiles) != 1 {
+		// default Config should only have default-scheduler configuration.
+		return v1beta2.PluginSet{}, xerrors.Errorf("get default scheduler configuration: %w", err)
+	}
+	return defaultConfig.Profiles[0].Plugins.Filter, nil
+}
+
+func AdditionalFilterPlugins() []v1beta2.Plugin {
```

Comment:
You call it `custom` in other places. Should we unify the notation?

```
func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
```

Reply:
Good catch. I think it's better to call them `outOfTreePlugins`